src/core/buffers/buffer_orchestrator.ts

```diff
@@ -424,7 +430,28 @@ export default function BufferOrchestrator(
           }))
         );
 
+    const needMediaSourceReloadAfterPeriodEnd$ =
+      manifest.getPeriodAfter(basePeriod) !== null &&
+      shouldReloadAtEachPeriodChange ?
+        endOfCurrentBuffer$.pipe(
+          mergeMap(() => {
+            return clock$.pipe(
+              mergeMap((clock) => {
+                if (basePeriod.end !== undefined && clock.currentTime >= basePeriod.end) {
+                  return observableOf({ type: "needs-media-source-reload" as const,
+                           value: { currentTime: clock.currentTime + 0.01,
+                                    isPaused: clock.isPaused } });
+                }
+                return EMPTY;
+              })
+            );
+          }),
+          take(1)
+        ) :
+        EMPTY;
```

Comment:
I get it but that's very difficult to read.

```js
const isLastPeriod = manifest.getPeriodAfter(basePeriod) === null;
const reloadAtPeriodEnd$ = isLastPeriod || !shouldReloadAtEachPeriodChange ?
  EMPTY :
  endOfCurrentBuffer$.pipe(
    mergeMapTo(clock$),
    mergeMap((tick) => {
      if (basePeriod.end === undefined || tick.currentTime < basePeriod.end) {
        return EMPTY;
      }
      const newTime = clock.currentTime + 0.01;
      return observableOf({ type: "needs-media-source-reload" as const,
                            value: { currentTime: newTime, isPaused: tick.isPaused }});
    }),
    take(1));
```

Reply:
Isn't there a problem with the `tick.currentTime < basePeriod.end` condition?

What if there is a small discontinuity at the end of the period?